desktop/apps/auction2/components/filter_sort/index.js

```diff
@@ -0,0 +1,54 @@
+import { default as React, PropTypes } from 'react';
+import { connect } from 'react-redux'
+import _ from 'underscore'
+import { updateSort } from '../../actions'
+
+function FilterSort({ sortMap, filterParams, updateSortAction }) {
+  const selectedSort = filterParams.sort
+  return (
+    <div className='auction2-filter-sort'>
+      <span className='auction2-filter-sort__label bordered-pulldown-label'>Sort by:</span>
+      <div className='bordered-pulldown'>
+        <a className='bordered-pulldown-toggle' href='#'>
+          <span className='bordered-pulldown-text'>{ sortMap[selectedSort] }</span>
+          <div className='bordered-pulldown-toggle-caret'>
+            <span className='caret'></span>
+          </div>
+        </a>
+        <div className='bordered-pulldown-options'>
+          {
+            _.map(sortMap, (sortName, sort) => {
```

Comment:
Ah! my bad -- I didn't realize you were iterating over an object.

@acjay - I think the comment you made a few weeks back about the "rule of least power" has seared itself into my cortex like a tattoo, which this brings to mind. The standard library in JS is indeed really limited, and so its often that you see lodash or underscore being used as a replacement (which I think is totally cool). One thing I always try to call out though is array iteration; seeing `_.each(myArray)` or `_.map(myOtherArray)` always brings to mind the period of time when underscore was used as a shim over missing IE features, and still to this day I see this practice continue almost unconsciously.